### PR TITLE
Refactor tests into individual functions

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

**Comment by maxwofford** [comment]
@zachlatta You will probably want to test this.

**Comment by zachlatta** [comment]
Before I review, just a quick heads up that you can include "Closes #13" in the pull request body to have the pull request automatically close #13 when it's merged.

**Comment by maxwofford** [comment]
@zachlatta rewording commit

**Comment by zachlatta** [comment]
I get the https://gist.github.com/zachlatta/008186572ad9e002cdf1 when running this on `hackedu/hackedu`. Can you look into it?

**Comment by maxwofford** [comment]
<s>Markcop takes bribes from code lines to _look the other way_</s>

MarkCop now ignores embedded code lines

File: bin/markcop [hunk 1]
```diff
@@ -41,155 +40,145 @@ cat <<'EOF'
...
-    printf "${GREEN}.${NC}"
-  fi
+    if [ $broken_link = true ]; then
+      errors="${errors}\n${file}:${line_num} '${link_dest}' is not a valid link!"
+    fi
+  done
+}
+
+for file in $(git ls-files | grep '\.md$'); do
+  content=$(grep -n "" $file | perl -0777 -pe 's/[0-9]+\:```.*?```//gs')
```
Comment (zachlatta) [inline bin/markcop]: Why the `?` in the regex?

Comment (maxwofford) [inline bin/markcop]: to keep it non-greedy:

``````
1. ```
2. code
3. ```
4. text
5. ```
6. code
7. ```
``````

The `?` ensures line 4 isn't selected

Comment (zachlatta) [inline bin/markcop]: Ah, gotcha. Important detail.

**Comment by maxwofford** [comment]
screenshot from 2015-11-02 00 05 03

**Comment by zachlatta** [comment]
Markcop is so pretty.

**Pull request merged**